pkg/convert/xliff12_test.go

```diff
@@ -74,61 +74,214 @@ func randXliff12(translation *model.Translation) []byte {
 func Test_FromXliff12(t *testing.T) {
 	t.Parallel()
 
-	t.Skip() // TODO
-
-	originalTranslation := testutilrand.ModelTranslation(
-		3,
-		[]testutilrand.ModelMessageOption{testutilrand.WithStatus(model.MessageStatusTranslated)},
-		testutilrand.WithOriginal(true),
-	)
-
-	nonOriginalTranslation := testutilrand.ModelTranslation(
-		3,
-		[]testutilrand.ModelMessageOption{testutilrand.WithStatus(model.MessageStatusUntranslated)},
-		testutilrand.WithOriginal(false),
-	)
+	//t.Skip() // TODO
+
+	//originalTranslation := testutilrand.ModelTranslation(
+	//	3,
+	//	[]testutilrand.ModelMessageOption{testutilrand.WithStatus(model.MessageStatusTranslated)},
+	//	testutilrand.WithOriginal(true),
+	//)
+	//
+	//nonOriginalTranslation := testutilrand.ModelTranslation(
+	//	3,
+	//	[]testutilrand.ModelMessageOption{testutilrand.WithStatus(model.MessageStatusUntranslated)},
+	//	testutilrand.WithOriginal(false),
+	//)
 
 	tests := []struct {
 		name     string
 		expected *model.Translation
 		data     []byte
 	}{
 		{
-			name:     "Original",
-			data:     randXliff12(originalTranslation),
-			expected: originalTranslation,
+			name: "input with x tag",
+			data: []byte(`<?xml version="1.0" encoding="UTF-8" ?>
+			<xliff version="1.2"
+				xmlns="urn:oasis:names:tc:xliff:document:1.2">
+			<file source-language="en">
+				<body>
+					<trans-unit id="9204248378636247318" datatype="html">
+						<source>Document <x id="PH" equiv-text="status.filename"/> was added to paperless.</source>
+ 					 	<context-group purpose="location">
+    					<context context-type="sourcefile">src/app/app.component.ts</context>
+    					<context context-type="linenumber">51</context>
+  						</context-group>
+  					<target state="needs-translation">Document <x id="PH" equiv-text="status.filename"/> was added to paperless.</target>
+					</trans-unit>
+				</body>
+			</file>	
+			</xliff>`),
+			expected: &model.Translation{
+				Original: true,
+				Language: language.English,
+				Messages: []model.Message{
+					{
+						ID: "9204248378636247318",
+						Message: ".local $x1 = { |<x id=\"PH\" equiv-text=\"status.filename\"/>| }\n" +
+							"{{Document { $x1 } was added to paperless.}}",
```

Comment:
Should it be a named placeholder? Does MF2 support variable name with `.` - `status.filename`? `{{Document { $status.filename } was added to paperless.}}`

Reply:
<img width="643" alt="image" src="https://github.com/translate-agent/translate/assets/94634373/19206853-df7a-490d-96ce-1cc3ebe4d1de">
This is an example of a test case from MF2 Builder. So It will work with a dot.
This test case was used from the PDF file that @VladislavsPerkanuks provided.

Reply:
we Could also make use of openning and closing functions. They are used for markup.

e.g.

```go
Message: "Document {+x :id=PH equiv-text=status.filename}{-x} was added to paperless"
```

but dunno about that 🤷

Reply:
What is the purpose of ` id=\"PH\"`? Does id contain a value from a canonical list?

Reply:
who knows :)

Reply:
Do we know where the example is coming from?